src/Xenial.Tasty/FeatureDetection/FeatureDetector.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Xenial.Delicious.FeatureDetection.Win;
+
+namespace Xenial.Delicious.FeatureDetection
+{
+    internal static class FeatureDetector
+    {
+        internal static readonly string[] CmdProcessNames = new[]
+        {
+            "cmd",
+            "VsDebugConsole",
+            "powershell"
+        };
+
+        internal static bool IsWindows()
+            => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+        public static bool IsCmd()
+        {
+            var proc = Process.GetCurrentProcess();
+
+            bool IsCmdProc(Process process)
+            {
+                if (process != null)
+                {
+                    if (CmdProcessNames.Any(name => string.Equals(name, process.ProcessName, StringComparison.CurrentCultureIgnoreCase)))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+            while(proc != null)
+            {
+                if(IsCmdProc(proc))
+                {
+                    return true;
+                }
+                proc = proc.Parent();
+            }
+            return false;
+        }
+
+        internal static bool IsWindowsTerminal()
+        {
+            var env = Environment.GetEnvironmentVariable("WT_SESSION");
+            if (!string.IsNullOrEmpty(env) && Guid.TryParse(env, out _))
```

Comment:
Guid.TryParse is safe for null and empty input

Reply:
Safe yes. But i think its not safe for nullable check

Reply:
```
    private static bool TryParseGuid(string g, Guid.GuidStyles flags, ref Guid.GuidResult result)
    {
      if (g == null)
      {
        result.SetFailure(Guid.ParseFailureKind.Format, "Format_GuidUnrecognized");
        return false;
      }
      string guidString = g.Trim();
      if (guidString.Length == 0)
      {
        result.SetFailure(Guid.ParseFailureKind.Format, "Format_GuidUnrecognized");
        return false;
      }
```

Reply:
But there is No annotation for the Compiler > [NotNull] so we have either to make a helper and so it on our own, or just live with the Double check until net5